Modifying Rating model and added tests.

ohadrevach opened the pull request:
Defined Rating model and added update function for it.
  Also added tests for adding,removing and updating rating.
Close #41 
Close #36

GalOved reviewed (changes requested):
Looking good to me good work.
If you look in the test_rating file you can see that you need too import a specific models. after that flake8 will pass.
And I would appreciate it if you shared with me why you chose to make static methods? Is there a reason for this that will help us in the future?
I don't think it's anything wrong, I just want to understand if I might also need to do it.

GalOved reviewed (approved):
Looking good to me

efraimvis reviewed (commented):
Please link this PR to an issue. Also, in case [this](https://github.com/redhat-beyond/tasties/pull/46) PR will get merged, I think you should consider removing 0001_initial.py from this PR.

natifridman reviewed (commented):
This PR is for the Rating model, do we need other models here as well?

natifridman commented on tasties_app/models.py:
```diff
@@ -1,3 +1,89 @@
+    )
+
+    @staticmethod
+    def add_rating(author_id, recipe_id, rating):
```
Why do you need this method if you don't use it?

natifridman commented on tasties_app/models.py:
```diff
@@ -1,3 +1,89 @@
+        rating.save()
+        return rating
+
+    def edit_rating(self, new_rate):
```
Maybe instead of `edit`, `update` will explain better what this method doing

natifridman commented on tasties_app/models.py:
```diff
@@ -1,3 +1,89 @@
+        self.rating = new_rate
+
+    @staticmethod
+    def remove_rating(rating):
```
Why do you need delete method if you can just do `rating.delete()`

natifridman commented on tasties_app/tests/test_rating.py:
```diff
@@ -0,0 +1,49 @@
+        assert rating.rating == 3
+
+    @pytest.mark.django_db
+    def test_delete(self, generate_data):
```
Please give the test a meaningful name, something like this
```suggestion
    def test_delete_rating(self, rating):
```

natifridman commented on tasties_app/tests/test_rating.py:
```diff
@@ -0,0 +1,49 @@
+        assert len(rating_set) == 0
+
+    @pytest.mark.django_db
+    def test_edit(self, generate_data):
```
Please give the test a meaningful name, something like this
```suggestion
    def test_edit_rating(self, rating):
```

GilShechter reviewed (approved):
LGTM, nice work

ohadrevach commented:
> This PR is for the Rating model, do we need other models here as well?

Yes, because Rating uses other models so it was needed to define them.

liornoy commented:
The commit description is a bit misleading: "Added functions for remov, adding and editing rating,"
Where you just added a function for updating the rating.
Also if you added all the other model classes, why not add the corresponding comments to everyone?
like you did for the rating class:
```
    """
    This Class represent Rating object
    Fields:
        author_id (User): User object.
        recipe_id (Recipe): Recipe object.
        rating (int): rating for the recipe, 1-5
    """
```

liornoy commented:
Also, this PR resolves issue #36, please add it to the PR description.

ohadrevach commented:
> The commit description is a bit misleading: "Added functions for remov, adding and editing rating," Where you just added a function for updating the rating. Also if you added all the other model classes, why not add the corresponding comments to everyone? like you did for the rating class:
> 
> ```
>     """
>     This Class represent Rating object
>     Fields:
>         author_id (User): User object.
>         recipe_id (Recipe): Recipe object.
>         rating (int): rating for the recipe, 1-5
>     """
> ```

Updated the description, I didn't write to anyone because everyone did this comments to the model he worked on.

liornoy commented:
Please reword the `Modifying Rating model` commit message, as you didn't add remove and add functions.

ohadrevach commented:
> Please reword the `Modifying Rating model` commit message, as you didn't add remove and add functions.

Done

liornoy commented:
You changed the PR title. I asked for the commit message of the "Modifying Rating model" commit.

ohadrevach commented:
> You changed the PR title. I asked for the commit message of the "Modifying Rating model" commit.

changed it back

natifridman commented on tasties_app/models.py:
```diff
@@ -1,3 +1,85 @@
+
+
+class Ingredient(models.Model):
+    WHOLE = "WHOLE"
+    FLOZ = "FLOZ"
+    TSP = "TSP"
+    OZ = "OZ"
+    CUP = "CUP"
+    GRAM = "GRAM"
+    ML = "ML"
+    UNIT_CHOICES = (
+        (WHOLE, "Whole"),
+        (FLOZ, "Fluid Ounce"),
+        (TSP, "Teaspoon"),
+        (OZ, "Ounce"),
+        (CUP, "Cup"),
+        (GRAM, "Gram"),
+        (ML, "Milliliter"),
+    )
```
Why not use enum for this?

natifridman reviewed (commented):
I would also add some tests here, what if I create a rating of 6 for example? or without an author?

liornoy commented:
> > You changed the PR title. I asked for the commit message of the "Modifying Rating model" commit.
> 
> changed it back

You still haven't updated that commit message.
In this PR you have 2 commits:
1. Populated models.py with model stubs, updated Pipfiles
2. Modifying Rating model 

The commit message of commit 2 is:
```
Added functions for remov, adding and editing rating,
Added tests for all of the above.
```

This message is not relevant as you didn't add "remove" and "adding" functions.
Therefore it's a misleading commit message.
Please reword this commit message using the git rebase reword command.

ohadrevach commented:
> > > You changed the PR title. I asked for the commit message of the "Modifying Rating model" commit.
> > 
> > 
> > changed it back
> 
> You still haven't updated that commit message. In this PR you have 2 commits:
> 
> 1. Populated models.py with model stubs, updated Pipfiles
> 2. Modifying Rating model
> 
> The commit message of commit 2 is:
> 
> ```
> Added functions for remov, adding and editing rating,
> Added tests for all of the above.
> ```
> 
> This message is not relevant as you didn't add "remove" and "adding" functions. Therefore it's a misleading commit message. Please reword this commit message using the git rebase reword command.

Understood, fixed it as requested

natifridman commented on tasties_app/tests/test_rating.py:
```diff
@@ -0,0 +1,46 @@
+    @pytest.mark.django_db
+    def test_edit_rating(self, rating):
+        rating_set = Rating.objects.all()
+        rating = rating_set[0]
+        rating.update_rating(4)
+        assert rating.rating == 4
```
Is this checking the DB? I see that you override `rating` and check the same object

liornoy replied:
I think what it does is test the update function and assert that it updates the rating field of the rating object.

ohadrevach replied:
I did what Lior said

natifridman replied:
Do we want to check that the update is saved?

ohadrevach replied:
Yes, i'm checking that this method is working fine

natifridman commented on tasties_app/tests/test_rating.py:
```diff
@@ -0,0 +1,56 @@
+            rating.update_rating(6)
+
+    @pytest.mark.django_db
+    def test_add_bad_rating_aid(self, rating):
```
It's unclear from the name, what is the difference between this test and the other test with the same name?

ohadrevach replied:
> It's unclear from the name, what is the difference between this test and the other test with the same name?

Added docstring for the functions

natifridman replied:
We usually try to make the test name self-explained. I think you can change the name to something like 
```suggestion
    def test_invalid_rating_author_id(self, rating):
```

natifridman reviewed (commented):
Please don't force push during the review, it makes it hard to understand what you changed from last time.

liornoy commented on tasties_app/models.py:
```diff
@@ -1,3 +1,82 @@
+    )
+
+    def update_rating(self, new_rate):
+        if type(new_rate) != int or new_rate > 5 or new_rate < 1:
+            raise ValueError("Invalid value")
```
This should be under the function's comment.

GalOved reviewed (approved):
LGTM

natifridman reviewed (approved):
LGTM, nice work!
